### PR TITLE
Implement Virtio driver, add pcap example, and add ixy-ci config

### PR DESCRIPTION
**Pull request by Bobo1239**
As stated in the title this adds a Virtio driver to ixy.rs. Also adds everything necessary to test ixy.rs with [ixy-ci](https://github.com/bobo1239/ixy-ci).

The development setup I've used is recorded in the `Vagrantfile`. For now only the [libvirt/qemu provider](https://github.com/vagrant-libvirt/vagrant-libvirt) `vagrant up --provider=libvirt ` should work as VirtualBox doesn't (or at least didn't) support `VIRTIO_F_ANY_LAYOUT`. Support for that can come at a later time...

I've tried to separate some changes into smaller commits so you can hopefully review this more easily.

I'll manually start an ixy-ci run on this PR for demonstration purposes. Later you'll have to add a GitHub webhook for this repository to start test runs.

### CODE REVIEW

**Comment by ackxolotl** [comment]
Very, very cool! I'll have a look at it in the next few days, I'm quite busy right now.

**Comment by Bobo1239** [comment]
Just fyi: I have a working branch with some changes required for VirtualBox but there's currently a humongous performance disparity to the C version which I haven't pinned down yet. My current plan is to eventually expose those changes via a cargo feature-flag but that will definitely have to wait for a separate PR.

File: ixy-ci.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+build = [
+    "sudo apt-get install -y curl gcc",
```
Comment (ackxolotl) [inline ixy-ci.toml:2]: Is `gcc` needed?

File: src/virtio.rs [hunk 2]
```diff
@@ -0,0 +1,671 @@
...
+    fn get_driver_name(&self) -> &str {
+        "ixy-virtio"
+    }
+
+    fn is_card_iommu_capable(&self) -> bool {
+        false
+    }
+
+    fn get_vfio_container(&self) -> Option<RawFd> {
+        panic!("VFIO isn't supported for Virtio");
```
Comment (ackxolotl) [inline src/virtio.rs]: Please return `None` here

**Review by ackxolotl — CHANGES_REQUESTED** [review]

**Comment by Bobo1239** [comment]
@ixy-ci ping

**Comment by leanderseidlitz** [comment]
ping @ixy-ci

**Comment by aevykk** [comment]
@ixy-ci ping

**Comment by Bobo1239** [comment]
@ixy-ci test

**Comment by Bobo1239** [comment]
@ixy-ci test
(Retrying with [higher SSH connection delay](https://github.com/Bobo1239/ixy-ci/commit/92975b674111ec3795e580b45dae93de3ce10899).)

**Review by ackxolotl — APPROVED** [review]

**Pull request merged**